k9mail/src/test/java/com/fsck/k9/mailstore/StoreSchemaDefinitionTest.java

```diff
@@ -295,11 +297,12 @@ private void assertDatabaseIndexesEquals(SQLiteDatabase expected, SQLiteDatabase
 
     private List<String> objectsInDatabase(SQLiteDatabase db, String type) {
         List<String> tables = new ArrayList<>();
-        Cursor cursor = db.rawQuery("SELECT name FROM sqlite_master WHERE type = ?", new String[] { type });
+        Cursor cursor = db.rawQuery("SELECT sql FROM sqlite_master WHERE type = ? AND sql IS NOT NULL", new String[] { type });
         try {
             if (cursor.moveToFirst()) {
                 while (!cursor.isAfterLast()) {
-                    tables.add(cursor.getString(cursor.getColumnIndex("name")));
+                    String sql = cursor.getString(cursor.getColumnIndex("sql"));
+                    tables.add("table".equals(type) ? sortTableColumns(sql) : sql);
```

Comment:
Let's not overengineer. Renaming `tables` to e.g. `databaseObjects` will align with the method name. Additionally, you could add a separate method for reading tables from the database. That will lead to some code duplication but reduces complexity inside the individual methods.